src/entry.rs

```diff
@@ -69,14 +69,13 @@ pub unsafe extern "C" fn secondary_entry(stack_end: *mut u64) -> ! {
         "isb",
         // Set the stack pointer which was passed.
         "mov sp, x0",
-        // Load Rust entry point address and argument from the bottom of the stack into
-        // callee-saved registers.
-        "ldp x19, x20, [sp, #-16]",
+        // Load the trampoline address into x20 and closure address into x0.
+        // This is loaded from StartCoreStack.
+        "ldr x20, [sp, #-16]",
+        "ldr x0, [sp, #-8]",
```

Comment:
Rather than having these offsets -8 and -16 as magic numbers in the assembly code, you could calculate them in a `const` parameter passed to the `naked_asm!`, something like `offset_of!(StartCoreStack, trampoline_ptr) - size_of::<StartCoreStack>()`.